lightning/src/ln/channelmanager.rs

```diff
@@ -2007,7 +2013,9 @@ where
 			return_malformed_err!("invalid ephemeral pubkey", 0x8000 | 0x4000 | 6);
 		}
 
-		let shared_secret = SharedSecret::new(&msg.onion_routing_packet.public_key.unwrap(), &self.our_network_key).secret_bytes();
+		let shared_secret = self.node_signer.ecdh(
+			Recipient::Node, &msg.onion_routing_packet.public_key.unwrap(), None
+		).unwrap().secret_bytes(); // TODO: Return error instead of unwrap?
```

Comment:
yeah, I'd lean towards erroring

Reply:
Yeah, that sounds reasonable. Also, should it fail immediately and back-propagate the error, or should it give the node signer the opportunity to sign again a bit later? Basically, what are the typical reasons it would fail, and would they auto-resolve?